specifyweb/businessrules/rules/tree_rules.py

```diff
@@ -2,13 +2,28 @@
 
 from specifyweb.businessrules.orm_signal_handler import orm_signal_handler
 from specifyweb.businessrules.exceptions import TreeBusinessRuleException
+from specifyweb.specify.tree_extras import is_instance_of_tree_def_item
+from specifyweb.specify.tree_ranks import *
 
 logger = logging.getLogger(__name__)
 
+@orm_signal_handler('pre_save')
```

Comment:
+1 on using an inheritance-based approach for instances of treedefitem! 
Will be even better when #4764 is implemented and IDEs can easily take you to the method definition. 

Although, if we ever intend on deleting treedefitems via `queryset.delete()` or through a cascade, then we should probably maintain `pre_delete`/`post_delete` signals: 

> Overridden model methods are not called on bulk operations
>
> Note that the delete() method for an object is not necessarily called when deleting objects in bulk using a QuerySet or as a result of a cascading delete. To ensure customized delete logic gets executed, you can use pre_delete and/or post_delete signals.

https://docs.djangoproject.com/en/3.2/topics/db/models/#overriding-model-methods

Reply:
Ya, I should of mentioned this in the comments.  My plan is use an inheritance approach after the #4358 issue is merged.

Reply:
wait, what's the difference in doing it right now, using inheritance, and doing it later, via inheritance, again? if you just use inheritance now, you can subclass later treeitems once https://github.com/specify/specify7/issues/4358 has been merged. BUT, you can also do it now.

Reply:
that is, my point is that there is no difference between doing it now, and once schema independence has been achieved.

Reply:
I have planned a follow-up issue to the Django schema issue that will do a lot of inheritance and other optimizations to the new models code.  I think it will be cleaner to do those changes in that PR.

Reply:
I don't believe it will be "cleaner". Can you expand on some specifics on why it will be? Curious about them. I think it'll be exactly same.

Trees work. they've been working for 10 years using inheritance, without schema independence, in a lot more aggressive environments like workbench. Anyways, won't block this PR bc of it.

Reply:
I went ahead and implemented an abstract TreeRank class with the save and delete logic override.  I'm keeping the pre_constraints_delete system around because it might be useful in the future.